File: SistFacturacion/src/main/java/paquete/Nevera.java
package paquete;

public class Nevera extends Electrodomestico{
    private double valorCapacidad;
    private double porcCapacidad;

    public Nevera( String opcImportado, Consumo consumo,double valorCapacidad){
        super(opcImportado, consumo);
        this.valorCapacidad = valorCapacidad;

    }

    public double getPorcCapacidad() {
        return porcCapacidad;
    }

    public void setPorcCapacidad(double porcCapacidad) {
        this.porcCapacidad = porcCapacidad;
    }

    public double getValorCapacidad() {
        return valorCapacidad;
    }

    public void setValorCapacidad(double valorCapacidad) {
        this.valorCapacidad = valorCapacidad;
    }

    public void calcularporcCapacidad(){
        if (getValorCapacidad()>120) {
            setPorcCapacidad((((getValorCapacidad() - 120) * 5) / 1000));
        } else{
            setPorcCapacidad(0);
        }
    }

    public double calcularCapacidad(){
        calcularporcCapacidad();
        setValorCapacidad(getTotalBase()*getPorcCapacidad());
        return getValorCapacidad();
    }
    @Override
    public  double calcularPrecioBase(){
        return getTotalBase()+calcularCapacidad();
    }




    }

File: Nevera.js
let Electrodomesticos = require('./Electrodomesticos');

module.exports = class Nevera extends Electrodomesticos{
    constructor(opcProcedencia, consumo, valorCapacidad){
        super(opcProcedencia, consumo);
        this.valorCapacidad = valorCapacidad;
        //this.porceCapacidad = porceCapacidad;
    }

    calcularporcCapacidad(){
        if (this.valorCapacidad>120) {
            this.porceCapacidad =(((this.valorCapacidad - 120) * 5) / 1000);
        } else{
            this.porceCapacidad = 0;
        }
    }

    calcularCapacidad(){
        this.calcularporcCapacidad();
        this.valorCapacidad=this.precio*this.porceCapacidad;
        return this.valorCapacidad;
    }

    calcularPrecioBase(){
        super.calcularPrecioBase();
        return this.precio+this.calcularCapacidad();
    }
}